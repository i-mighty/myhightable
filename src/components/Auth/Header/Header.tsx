import React from 'react';
import styled from 'styled-components/native';
import { Header as NBHeader, Text, Left, Body } from 'native-base';
import colorOpacity from '@src/utils/color-opacity';
import Banner from '@src/assets/highTableLogo.svg';

interface HeaderProps {
  testID?: string;
}

const StyledHeader = styled(NBHeader)<HeaderProps>`
  background-color: ${({ theme }) => colorOpacity(theme.vars.dark, 0.5)};
  width: 100%;
`;

const ButtonText = styled(Text)`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  text-align: center;
  color: ${({ theme }) => colorOpacity(theme.vars.white, 0.5)};
`;

const Header: React.FC<HeaderProps> = ({ testID }) => {
  return (
    <StyledHeader testID={testID} transparent noLeft>
      <Body>
        <Banner width={149} height={38} testID="banner" />
      </Body>
    </StyledHeader>
  );
};

export default Header;
