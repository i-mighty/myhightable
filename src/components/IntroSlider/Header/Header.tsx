import React from 'react';
import styled from 'styled-components/native';
import { Header as NBHeader, Button, Right, Text } from 'native-base';
import colorOpacity from '@src/utils/color-opacity';

interface HeaderProps {
  buttonText?: string;
  onButtonClick?: () => void;
  testID?: string;
  buttonTestID?: string;
}

const StyledHeader = styled(NBHeader)<HeaderProps>`
  background-color: ${({ theme }) => colorOpacity(theme.vars.dark, 0.5)};
  width: 100%;
`;

const ButtonText = styled(Text)`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  text-align: center;
  color: ${({ theme }) => colorOpacity(theme.vars.white, 0.5)};
`;

const Header: React.FC<HeaderProps> = ({
  buttonText,
  onButtonClick,
  testID,
  buttonTestID,
}) => {
  return buttonText ? (
    <StyledHeader testID={testID} transparent>
      <Right>
        <Button testID={buttonTestID} transparent onPress={onButtonClick}>
          <ButtonText>{buttonText}</ButtonText>
        </Button>
      </Right>
    </StyledHeader>
  ) : (
    <StyledHeader testID={testID} transparent />
  );
};

export default Header;
