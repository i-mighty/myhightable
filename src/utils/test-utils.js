import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
