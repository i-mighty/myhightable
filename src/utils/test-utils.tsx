import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
