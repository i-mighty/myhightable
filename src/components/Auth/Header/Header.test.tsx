import Header from './Header';
import { renderWithProps } from '@src/utils/test-utils';
import { fireEvent, NativeTestEvent } from '@testing-library/react-native';

const onButtonClick = jest.fn(() => {
  console.log('Pressed');
});

const defaultProps = { testID: 'header' };

describe('Header', () => {
  it('renders properly ', () => {
    const { getByTestId, baseElement } = renderWithProps(Header, defaultProps);
    expect(getByTestId(defaultProps.testID)).toBeDefined();
    expect(getByTestId('banner')).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });
});
