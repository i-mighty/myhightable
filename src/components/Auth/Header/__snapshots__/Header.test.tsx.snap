// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly  1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
          }
        }
      >
        <View
          noLeft={true}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "transparent",
                "borderBottomWidth": 0.5,
                "elevation": 0,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 6,
                "paddingRight": 10,
                "paddingTop": 18,
                "right": 0,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
                "top": 0,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "width": "100%",
              },
            ]
          }
          testID="header"
          transparent={true}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              }
            }
          >
            <SvgMock
              height={38}
              testID="banner"
              width={149}
            />
          </View>
        </View>
      </SafeAreaView>
    </View>
  </View>
</View>
`;
